test/m4/appframework/appframework_suite_test.go

```diff
@@ -63,21 +65,26 @@ var _ = BeforeSuite(func() {
 	testenvInstance, err = testenv.NewDefaultTestEnv(testSuiteName)
 	Expect(err).ToNot(HaveOccurred())
 
-	// create Valid Apps1
+	// Create Valid apps
 	appListV1 = testenv.BasicApps
+	appFileListV1 = make([]string, 0, len(appListV1))
+	for _, app := range appListV1 {
+		appFileListV1 = append(appFileListV1, testenv.AppInfo[app]["V1filename"])
+	}
 
 	// Download V1 Apps from S3
-	err = testenv.DownloadFilesFromS3(testDataS3Bucket, s3AppDirV1, downloadDirV1, appListV1)
+	err = testenv.DownloadFilesFromS3(testDataS3Bucket, s3AppDirV1, downloadDirV1, appFileListV1)
 	Expect(err).To(Succeed(), "Unable to download V1 app files")
 
-	// Parse ValidAppsV1 to create Valid Apps2
-	appListV2 = make([]string, 0, len(appListV1))
-	for _, app := range appListV1 {
-		appListV2 = append(appListV2, testenv.AppInfo[app]["V2filename"])
+	// Parse AppListV1 to create Valid Apps2
+	appListV2 = append(appListV1, testenv.NewAppsAddedBetweenPolls...)
+	appFileListV2 = make([]string, 0, len(appListV2))
```

Comment:
We can add a method in appframework util which return files name when give a list of app names like:
```
func GetAppFileNames(appList []string, version string) fileNames []string {
   var appFileList []string
    for _, app := range appList {
		appFileList = append(appFileList, AppInfo[app][version])
	}
  return appFileList
}
```
This would eliminate the need to keep 2 different lists.

Reply:
Since the `appFilelist` is needed in both before suite(to download and upload files) and the test(to validate download). Would we not still need to keep the 2 lists.

Or should we generate every-time we need it?

Reply:
We can generate it every time in method call or in verification methods.